rocketpy/EnvironmentAnalysis.py

```diff
@@ -594,22 +596,99 @@ def calculate_maximum_wind_gust(self):
+            density=True,
+            histtype="stepfilled",
+            alpha=0.2,
+            label="Wind Gust Speed Distribution",
+        )
+
+        # Plot weibull distribution
+        c, loc, scale = stats.weibull_min.fit(self.wind_gust_list)
+        x = np.linspace(0, np.max(self.wind_gust_list), 100)
+        plt.plot(
+            x,
+            stats.weibull_min.pdf(x, c, loc, scale),
+            "r-",
+            linewidth=2,
+            label="Weibull Distribution",
+        )
+
+        # Label plot
+        plt.ylabel("Probability")
+        plt.xlabel("Wind gust speed (m/s)")
+        plt.title("Wind Gust Speed Distribution")
+        plt.legend()
+        plt.show()
+
+        return self.wind_gust_list
 
     # TODO: Implement
     def calculate_average_temperature_along_day(self):
         """temperature progression throughout the day at some fine interval (ex: 2 hours) with 1, 2, 3, sigma contours"""
         ...
 
-    # TODO: Implement
-    def calculate_average_wind_profile(self):
-        """average, 1, 2, 3 sigma wind profile from 0 35,000 ft AGL"""
-        ...
+    # TODO: Create tessts
+    def plot_average_wind_speed_profile(self):
+        """Average wind speed for all datetimes available."""
+        altitude_list = np.linspace(1495.155, 12000, 100)
+        wind_speed_profiles = [
+            dayDict[hour]["windSpeed"](altitude_list)
+            for dayDict in self.pressureLevelDataDict.values()
+            for hour in dayDict.keys()
+        ]
+        self.average_wind_speed_profile = np.mean(wind_speed_profiles, axis=0)
+        # Plot
+        plt.figure()
+        plt.plot(self.average_wind_speed_profile, altitude_list, "r")
+        plt.plot(
+            np.percentile(wind_speed_profiles, 50 - 34.1, axis=0),
+            altitude_list,
+            "b--",
+            alpha=1,
+        )
+        plt.plot(
+            np.percentile(wind_speed_profiles, 50 + 34.1, axis=0),
+            altitude_list,
+            "b--",
+            alpha=1,
+        )
+        plt.plot(
+            np.percentile(wind_speed_profiles, 50 - 47.4, axis=0),
+            altitude_list,
+            "b--",
+            alpha=0.5,
+        )
+        plt.plot(
+            np.percentile(wind_speed_profiles, 50 + 47.7, axis=0),
+            altitude_list,
+            "b--",
+            alpha=0.5,
+        )
+        # plt.plot(np.percentile(wind_speed_profiles, 50-49.8, axis=0, method='weibull'), altitude_list, 'b--', alpha=0.25)
+        # plt.plot(np.percentile(wind_speed_profiles, 50+49.8, axis=0, method='weibull'), altitude_list, 'b--', alpha=0.25)
+        for wind_speed_profile in wind_speed_profiles:
+            plt.plot(wind_speed_profile, altitude_list, "gray", alpha=0.01)
+        plt.xlabel("Wind speed (m/s)")
+        plt.ylabel("Altitude (m)")
+        plt.title("Average Wind Speed Profile")
+        plt.show()
+
+        return self.average_wind_speed_profile
```

Comment:
```suggestion
        return None
```

I suggest we do not return the array since we are already ploting, otherwise look what could happen: 

![image](https://user-images.githubusercontent.com/63590233/168718658-61b74fb9-b886-4abe-a039-a63e01a48c8b.png)